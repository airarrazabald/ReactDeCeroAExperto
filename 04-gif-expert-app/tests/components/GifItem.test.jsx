import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem/>', () => {
    
    const title = 'One Piece';
    const url = 'http://testing.cl/one-piece.png';

    test('debe hacer match con el snapshot', () => { 
        const { container } =render(<GifItem title={ title } url={ url }  />);

        expect( container ).toMatchSnapshot();
        
     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={ title } url={ url }  />);

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe(url);
        expect( alt ).toBe(title);
     });

     test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={ title } url={ url }  />);
        
        expect( screen.getByText( title )).toBeTruthy();
     });
 });