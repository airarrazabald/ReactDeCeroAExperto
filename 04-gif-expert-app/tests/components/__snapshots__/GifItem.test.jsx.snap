// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="http://testing.cl/one-piece.png"
    />
    <p>
      One Piece
    </p>
  </div>
</div>
`;
